Tests/SwiftDocCTests/Utility/ValidatedURLTests.swift

```diff
@@ -81,6 +81,14 @@ class ValidatedURLTests: XCTestCase {
     }
     
     func testQueryIsPartOfPathForAuthoredLinks() throws {
+        
+        func validate(linkText: String, expectedPath: String, expectedFragment: String? = nil) throws {
+            let validated = try XCTUnwrap(ValidatedURL(parsingAuthoredLink: linkText), "Failed to parse \(linkText.singleQuoted) as authored link")
+            XCTAssertNil(validated.components.queryItems, "Authored documentation links don't include query items")
+            XCTAssertEqual(validated.components.path, expectedPath)
+            XCTAssertEqual(validated.components.fragment, expectedFragment)
+        }
```

Comment:
It's worth adding file/line arguments for assertion wrappers:

```suggestion
        func validate(linkText: String, expectedPath: String, expectedFragment: String? = nil, file: StaticString = #file, line: UInt = #line) throws {
            let validated = try XCTUnwrap(ValidatedURL(parsingAuthoredLink: linkText), "Failed to parse \(linkText.singleQuoted) as authored link", file: file, line: line)
            XCTAssertNil(validated.components.queryItems, "Authored documentation links don't include query items", file: file, line: line)
            XCTAssertEqual(validated.components.path, expectedPath, file: file, line: line)
            XCTAssertEqual(validated.components.fragment, expectedFragment, file: file, line: line)
        }
```

This way, assertion failures point to the caller of validate rather than the body.

Reply:
Thanks for this!